contrib/devtools/github-merge.py

```diff
@@ -152,67 +188,111 @@ def parse_arguments():
 
 def main():
     # Extract settings from git repo
+    host_default_github='git@github.com'
     repo = git_config_get('githubmerge.repository')
-    host = git_config_get('githubmerge.host','git@github.com')
+    host = git_config_get('githubmerge.host', host_default_github)
     opt_branch = git_config_get('githubmerge.branch',None)
     testcmd = git_config_get('githubmerge.testcmd')
     signingkey = git_config_get('user.signingkey')
+
+    baseurl_gitlab = git_config_get('githubmerge.gitlabbaseurl')
+
     if repo is None:
         print("ERROR: No repository configured. Use this command to set:", file=stderr)
         print("git config githubmerge.repository <owner>/<repo>", file=stderr)
+        print("if you want to use gitlab instead github, then in addition also set URL (of github.com or own server) with command:", file=stderr)
+        print("git config githubmerge.gitlabbaseurl https://gitlab.com/", file=stderr)
         sys.exit(1)
     if signingkey is None:
         print("ERROR: No GPG signing key set. Set one using:",file=stderr)
         print("git config --global user.signingkey <key>",file=stderr)
         sys.exit(1)
 
+    api_mode='github'
+    if baseurl_gitlab is not None:
+        api_mode='gitlab'
+
+    if (api_mode == 'gitlab') and (host == host_default_github) :
```

Comment:
should be nxor instead of and?

Reply:
Hm, why?

This logic is that when you are in gitLAB mode, but the host/ssh is the one from gitHUB, then you are mistaken.
(and this is common case befcause host/ssh for gitHUB is the default if unset),
other cases are ok:

mode==gitLAB,  ssh==gitHUB  --> this error message, seems a mistake
mode==gitLAB,  ssh==git@gitLAB.yourserver.com etc  --> ok
mode==gitHUB,  ssh==gitHUB --> ok, the normal old default behaviuour
mode==gitHUB,  ssh==git@git.yourgitHUBapi-server.com --> ok. well dunno if it's possible to deploy github API server (is it opensource?) but that was always allowed to specify that option